File: RA1/JAVA/EjemploGatos/Gato.java
package EjemploGatos;

public class Gato{
    //Atributos
    private String name;
    private String sex;
    private int age;
    private int weight;
    private String color;
    private String texture;

    //Métodos
    //Constructor
    //public Gato(){}
    public Gato(String name, String sex, int age, int weight, String color, String texture){
        //clausula "this", indica que la variable es un atributo del objeto
        this.name = name;
        this.sex = sex;
        this.age = age;
        this.weight = weight;
        this.color = color;
        this.texture = texture;
    }

    public String darInformacion(){
        return "Información Gato:\n\tname: " + this.name + "\n\tsex: " + this.sex + "\n\tage: " + this.age + "\n\tweight: " + this.weight + "\n\tcolor: " + this.color + "\n\ttexture: " + this.texture;
    }

    public void addAge(){
        this.age++;
    }

    public void changeColor(String newColor){
        String[] colores = {"blanco", "marrón", "gris", "bicolor", "rojizo", "rubio"};
        boolean aceptedColor = false;
        for(String color: colores){
            if(newColor.equals(color))
                aceptedColor = true;
        }
        if(aceptedColor)
            this.color = newColor;
        else
            System.out.println("El color no es aceptado");
    }
}

File: RA1/PYTHON/ejemplo_gatos/gato.py
class Gato:
    #atributos
    #No es necesaria

    #métodos
    #Constructor
    def __init__(self, name, sex, age, weight, color, texture):
        # atributos
        # SELF se refiere a los atributos propios de un objeto
        #VISIBILIDAD
        #<nombre_variable> = public
        #_<nombre_variable> = protegido (acuerdo de desarrolladores)
        #__<nombre_variable> = privado (genera un error en la interpretacion)
        self.__name = name
        self.__sex = sex
        self.__age = age
        self.__weight = weight
        self.__color = color
        self.__texture = texture

    def dar_informacion(self):
        return "Información Gato: \n\tname: " + self.__name + "\n\tsex: " + self.__sex + "\n\tage: " + str(self.__age) + "\n\tweight: " + str(self.__weight) + "\n\tcolor: " + self.__color + "\n\ttexture: " + self.__texture
    
    def add_age(self):
        self.__age += 1

    def change_color(self, new_color):
        colores = ["blanco", "marrón", "gris", "bicolor", "rojizo", "rubio"]
        acepted_color = False

        for color in colores:
            if new_color == color:
                acepted_color = True
        
        if acepted_color:
            self.__color = new_color
        else:
            print("El color no es aceptado")
